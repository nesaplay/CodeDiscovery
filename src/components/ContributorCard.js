import React from 'react'
import { Card, Image, Button } from 'semantic-ui-react'

const ContributorCard = ({
	avatar_url,
	contributions,
	login,
	type,
	html_url
}) => {
	const { Content, Header, Meta, Description } = Card

	return (
		<Card fluid color="orange">
			<Content>
				<Image
					floated="right"
					src={avatar_url}
					size="tiny"
					style={{ position: 'absolute', right: '20px' }}
				/>
				<Header>{login}</Header>
				<Meta>{type}</Meta>
				<Description>{contributions} contributions</Description>
			</Content>
			<Content extra>
				<div className="ui buttons fluid">
					<a href={html_url} target="_blank" style={{ width: '100%'}}>
						<Button fluid basic color="orange">
							More from this user
						</Button>
					</a>
				</div>
			</Content>
		</Card>
	)
}

export default ContributorCard
