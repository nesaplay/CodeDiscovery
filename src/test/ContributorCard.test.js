import React from 'react'
import ContributorCard from '../components/ContributorCard'

it('renders a component', () => {
	const wrapper = shallow(
        <ContributorCard />
    )

	expect(wrapper).toMatchSnapshot()
})
