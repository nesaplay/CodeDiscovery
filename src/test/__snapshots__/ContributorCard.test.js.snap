// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a component 1`] = `
<Card
  color="orange"
  fluid={true}
>
  <CardContent>
    <Image
      as="img"
      floated="right"
      size="tiny"
      style={
        Object {
          "position": "absolute",
          "right": "20px",
        }
      }
      ui={true}
    />
    <CardHeader />
    <CardMeta />
    <CardDescription>
       contributions
    </CardDescription>
  </CardContent>
  <CardContent
    extra={true}
  >
    <div
      className="ui buttons fluid"
    >
      <a
        style={
          Object {
            "width": "100%",
          }
        }
        target="_blank"
      >
        <Button
          as="button"
          basic={true}
          color="orange"
          fluid={true}
        >
          More from this user
        </Button>
      </a>
    </div>
  </CardContent>
</Card>
`;
